types/styled-system/index.d.ts

```diff
@@ -96,13 +96,32 @@ export interface VariantArgs {
 
 export function variant(props: VariantArgs): (...args: any[]) => any;
 
+export type ObjectOrArray<T> = T[] | { [K: string]: T };
+
 export interface BaseTheme {
     breakpoints?: number[];
     space?: number[];
     fontSizes?: number[];
-    colors?: {
-        [name: string]: string;
-    };
+    colors?: ObjectOrArray<CSS.ColorProperty>;
+}
+
+export interface Theme extends BaseTheme {
+    fonts?: ObjectOrArray<CSS.FontFamilyProperty>;
+    fontWeights?: ObjectOrArray<CSS.FontWeightProperty>;
+    maxWidths?: ObjectOrArray<CSS.WidthProperty<{}>>;
+    minWidths?: ObjectOrArray<CSS.WidthProperty<{}>>;
+    heights?: ObjectOrArray<CSS.HeightProperty<{}>>;
+    maxHeights?: ObjectOrArray<CSS.HeightProperty<{}>>;
+    minHeights?: ObjectOrArray<CSS.HeightProperty<{}>>;
+    lineHeights?: ObjectOrArray<CSS.LineHeightProperty<{}>>;
+    letterSpacings?: ObjectOrArray<CSS.LetterSpacingProperty<{}>>;
+    shadows?: ObjectOrArray<CSS.BoxShadowProperty>;
+    borders?: ObjectOrArray<CSS.BorderProperty<{}>>;
+    radii?: ObjectOrArray<CSS.BorderRadiusProperty<{}>>;
+    opacity?: ObjectOrArray<CSS.GlobalsNumber>;
+    textStyles?: ObjectOrArray<CSS.StandardProperties>;
+    colorStyles?: ObjectOrArray<CSS.StandardProperties>;
+    buttons?: ObjectOrArray<CSS.StandardProperties>;
```

Comment:
# minor but alphabatizing would make these easier to scan

Reply:
💚 7c243ec